web/base_tree_viewer.js

```diff
@@ -71,7 +71,9 @@ class BaseTreeViewer {
   /**
    * Prepend a button before a tree item which allows the user to collapse or
    * expand all tree items at that level; see `_toggleTreeItem`.
-   * @private
+   * @protected
+   * @param {Element} div
+   * @param {any} [hidden=false]
```

Comment:
```suggestion
   * @param {HTMLDivElement} div
   * @param {boolean} [hidden]
   * @protected
```

Reply:
this method's signature is different for child classes (they are passing an object as a second parameter). In typescript this is not allowed, that's why I set hidden to any as a workaround.

Reply:
No, they are not:
 - https://github.com/mozilla/pdf.js/blob/d8331e31005edba6afef28ff2f4d55ffe5731df4/web/pdf_outline_viewer.js#L185
 - https://github.com/mozilla/pdf.js/blob/d8331e31005edba6afef28ff2f4d55ffe5731df4/web/pdf_layer_viewer.js#L103

Hence the changes I mentioned are correct here for *the base* method.

Reply:
the call to parent class is correct, but I am talking about the child classes' redefined method signature
https://github.com/mozilla/pdf.js/blob/d8331e31005edba6afef28ff2f4d55ffe5731df4/web/pdf_outline_viewer.js#L167

Reply:
here is a simple demo, just for case:

https://www.typescriptlang.org/play?#code/MYGwhgzhAEAKYCcCmA7ALtA3gKGtAtkmgBYD2AJgBSIDmAXNCgK74BGSCAlFtrtAL7ZB2UJBgBhYgEsQ5aEgAeaVORjxk6HnkIkK1BPWitSpEEjApuOPHj6D+QA